src/datasources/tag/components/TagQueryEditor.tsx

```diff
@@ -28,6 +29,23 @@ export function TagQueryEditor({ query, onChange, onRunQuery, datasource }: Prop
     onRunQuery();
   };
 
+  const getWorkspaceOptions = () => {
+    if (!workspaces.loading) {
+      const options = workspaces.value;
+      options?.unshift(...getVariableOptions())
+      return options;
+    }
+    else {
+      return [];
+    }
+  }
+
+  const getVariableOptions = () => {
+    return getTemplateSrv()
+      .getVariables()
+      .map((variable) => ({ label: '$' + variable.name, value: '$' + variable.name }));
+  };
+
```

Comment:
I think it would be nice if we could add this logic to the `useWorkspaceOptions` hook, so we can reuse it. 
We should add `templateSrv` to the `DataSourceBase` constructor and then `useWorkspaceOptions` can call into it to get the variables. It'll be a bit cleaner too, because we won't need the `if (workspaces.loading)` check.

Reply:
Done. This PR is ready for re-review.